### DISCUSSION TITLE
USDC gas payment feature

### DISCUSSION THREAD

**Discussion by trinitys7** [akash-network/community#335] [General]
The topic of this discussion is about implementing the USDC gas payment feature. Currently, there're couple of options.
1. Using the fee abs module
    This means all the USDC tx fee will be convert to AKT via ibc swap on osmosis. This module use async-ICQ to query spot price infomation in osmosis. 
    
    For query price from osmosis, fee abs will send a IBC packet to osmosis with port is , the async-icq module in osmosis handle this packet and send the result back to fee abs module by ack packet that include price data.
![](https://hackmd.io/_uploads/rJrQ-3dZ6.png)  
    - Pros: 
        - Can be expand to allow paying  other types of currency.
        - Community pool will only have one currency which is Akash, thus easier to manage.
    - Cons: 
        - High complexity, the logic flow is not easily understandable.
        - Put more weight on the chain code base, more troublesome maintainence.
        - Depended in relayer and osmosis, currently in fee abs, when fee abs module receive failed ACK packet from osmosis in async-icq, the USDC fees features will be disable.
        

2. Configuring akash nodes allowing USDC transaction gas
    This means we accept the usdc as gas and keeping the usdc gas as is.
    - Pros: 
        - Simple and easy, no coding requires
    - Cons:
        - These fees will be distributed for staking rewards and the community pool like normal tx fees. It is not reasonable because each user would only receive a very little amount of USDC due to the low total amount of all USDC fees. It can make those USDC fees wasted.
        - We can modify the current fees process all for make these fees only sent to the community pool. 

3. Enabling zero fees transactions for USDC deployments
    - Pros:
        - Simple and easy to implement.
    - Cons:
        - Due to the absence of transaction fees, the blockchain may be vulnerable to DDOS attacks.

**Comment by jommi9** [comment]
Definitely a step forward imho.
